src/components/File/File.tsx

```diff
@@ -0,0 +1,59 @@
+import * as classNames from 'classnames';
+import * as React from 'react';
+import { rivetize, shortuid } from '../Rivet';
+
+interface FileProps {
+    fileName?: string;
+    label?: string;
+}
+
+const initialState = { files: '' }
+type FileState = Readonly<typeof initialState>
+
+const FileInput: React.SFC<FileProps & React.HTMLAttributes<HTMLInputElement>> = ({ className, fileName, id = shortuid(), label = 'Upload a file', ...attrs }) => (
+    <div className={classNames('rvt-file', className)}>
+        <input {...attrs} type="file" id={id} aria-describedby={id + "-file-description"} />
+        <label htmlFor={id} className="rvt-button">
+            <span>{label}</span>
+            <svg role="img" xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
+                <path fill="currentColor" d="M10.41,1H3.5A1.3,1.3,0,0,0,2.2,2.3V13.7A1.3,1.3,0,0,0,3.5,15h9a1.3,1.3,0,0,0,1.3-1.3V4.39ZM11.8,5.21V6H9.25V3h.34ZM4.2,13V3h3V6.75A1.25,1.25,0,0,0,8.5,8h3.3v5Z" />
+            </svg>
+        </label>
+        <div className="rvt-file__preview" id={id + "-file-description"}>
+            { fileName ? <span>{fileName}</span> : 'No file selected' }
+        </div>
+    </div>
+);
+FileInput.displayName = 'FileInput';
+
+class File extends React.PureComponent<FileProps & React.HTMLAttributes<HTMLInputElement>, FileState> {
```

Comment:
You may notice I put the stateful and presentational component for the file input in the same file here.  My reasoning for this is that unlike the `Alert` this component felt like one which should always have this behavior.  I have exported the presentational version of the component just in case, but it would not be bundled in the normal components list.  If someone has a need they should be able to import it independently I believe.